src/Login.tsx

```diff
@@ -1,9 +1,109 @@
-import React from "react";
+import React, { useState } from "react";
+import { useNavigate } from "react-router";
+import axios from "axios";
 
-const Login = () => (
-	<div>
-		<h1>Login Page</h1>
-	</div>
-);
+type FormFieldsType = {
+	email: { value: string };
+	password: { value: string };
+} & EventTarget;
+
+type SafeUser = {
+	id: number;
+	firstName: string;
+	lastName: string;
+	email: string;
+};
+
+const Login = () => {
+	const navigate = useNavigate();
+	const [error, setError] = useState<string>();
+	const [loginStatus, setLoginStatus] = useState(false);
+	const [loggedInUser, setLoggedInUser] = useState<SafeUser>();
+
+	const isValid = (e: React.FormEvent<HTMLFormElement>): boolean => {
+		const target = e.target as FormFieldsType;
+
+		const emailInput = target.email.value;
+		const passwordInput = target.password.value;
+		if (emailInput === "" || passwordInput === "") {
+			setError("some fields are incomplete");
+		} else if (passwordInput.length < 8) {
+			setError("password must be more than 8 characters");
+		} else if (emailInput.includes("@") === false) {
+			setError("Please enter a valid email");
+		} else {
+			return true;
+		}
+		return false;
+	};
+	const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+		e.preventDefault();
+
+		const target = e.target as FormFieldsType;
+		const email = target.email.value;
+		const password = target.password.value;
+
+		if (isValid(e)) {
+			axios
+				.post("http://localhost:8080/customers", {
+					email,
+					password,
+				})
+				.then(response => {
+					setLoginStatus(true);
+					setLoggedInUser(response.data);
+					navigate("/profile");
+				})
+				.catch(errors => {
+					console.log("errors: ", errors);
+					if (errors.toJSON().message === "Network Error") {
+						setError("Error... Something went wrong");
+					}
+				});
```

Comment:
doesn't look like we're handling 400s here

Reply:
this has been set up for getting the response error